graph/db/addr.go

```diff
@@ -150,6 +150,43 @@ func encodeOpaqueAddrs(w io.Writer, addr *lnwire.OpaqueAddrs) error {
 	return err
 }
 
+// encodeDNSAddr serializes a DNS address into its compact raw bytes
+// representation. It writes the address type, hostname length, hostname, and
+// port (in big-endian order) to the writer. The function validates that the
+// DNS address. Returns an error if any part of the serialization fails.
+func encodeDNSAddr(w io.Writer, addr *lnwire.DNSAddr) error {
```

Comment:
as discussed offline & in previous review comment: it is going to be much simpler to not touch the persistence layer. Rather just dynamically extract DNS addrs at read time from the existing opaque type & similarly at write time, add any DNS types to the opaque stream.

Reply:
because also, note that as it currently stands: any existing DNS addrs that we have stored already will currently be under the opaque addr type - so you have to account for those at read-time anyways.